src/service/resource.js

```diff
@@ -74,8 +74,9 @@ export class Resource {
    * @return {!Resource}
    */
   static forElement(element) {
-    return dev().assert(Resource.forElementOptional(element),
-        'Missing resource prop on %s', element);
+    return /** @type {!Resource} */ (
+        dev().assert(Resource.forElementOptional(element),
```

Comment:
Type inference seems to be having issues in general. Not sure exactly what
is going on.

On Fri, Sep 9, 2016 at 4:29 PM, erwin mombay <notifications@github.com>
wrote:

> In src/service/resource.js
> <https://github.com/ampproject/amphtml/pull/4908#discussion_r78263228>:
>
> > @@ -74,8 +74,9 @@ export class Resource {
> >     * @return {!Resource}
> >     */
> >    static forElement(element) {
> > -    return dev().assert(Resource.forElementOptional(element),
> > -        'Missing resource prop on %s', element);
> > +    return /** @type {!Resource} */ (
> > +        dev().assert(Resource.forElementOptional(element),
>
> yeah type inference system for asserts dont work with dev().assert(). no
> quick solution yet
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4908/files/9dc6e433e9326d421095feb7f85465681e43f296#r78263228>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeT3b9_Zxya-0DqG_4Qwgon8hre5vVks5qoevSgaJpZM4J5hjb>
> .
>